xarray/core/dataarray.py

```diff
@@ -3230,6 +3230,170 @@ def map_blocks(
 
         return map_blocks(func, self, args, kwargs)
 
+    def pad(
+        self,
+        pad_width: Mapping[Hashable, Union[int, Tuple[int, int]]] = None,
+        mode: str = "constant",
+        stat_length: Union[
+            None, int, Tuple[int, int], Mapping[Hashable, Tuple[int, int]]
+        ] = None,
```

Comment:
FWIW you don't need to supply `None` in the union if the default is `None` (e.g. https://github.com/pydata/xarray/pull/3596/files#diff-ffd3597671590bab245b3193affa62b8R3246)